angular-project/src/app/card-tracker/card-tracker.component.ts

```diff
@@ -0,0 +1,38 @@
+import { Component, Input, OnInit } from '@angular/core';
+import TimeData from 'src/assets/json/data.json';
+import { TimeInterface } from 'src/app/time-data-interface';
+
+@Component({
+  selector: 'app-card-tracker',
+  templateUrl: './card-tracker.component.html',
+  styleUrls: ['./card-tracker.component.scss']
+})
+export class CardTrackerComponent implements OnInit {
+  @Input() currentPeriod: string = 'daily' // initial value 'daily
+  @Input() currentActivity: TimeInterface = TimeData[0]; // initial value TimeData[0]
+  // get hours for selected time period
+  getHours (period:string) {
+    if (period === 'daily')
+      return this.currentActivity.timeframes.daily;
+    else if (period === 'weekly')
+      return this.currentActivity.timeframes.weekly;
+    else // (period === 'monthly')
+      return this.currentActivity.timeframes.monthly;
+  }
+  // transform word to noun for previous-period element
+  periodToNoun (period: string): string {
+    if (period === 'daily') return 'day';
+    else if (period === 'weekly') return 'week';
+    else return 'month';
+  }
```

Comment:
здесь есть 3 возможных варианта:
1. `switch case` 
2. создать свойство компонента, которое будет однозначно интерпретировать название периода:
```
prop = { 
  daily: 'day',
  weekly: 'week',
  monthly: 'month'
}
```
И вызвать в методе:
```suggestion
  periodToNoun (period: string): string {
    return this.prop[period];
  }
```
3. Так как язык - английский, то отсечь последние две буквы строки:
```suggestion
  periodToNoun (period: string): string {
    return period.slice(0, -2)
  }
```

Мне лично нравится 2 вариант

Reply:
Никита, привет! Я попробовал осуществить второй вариант, но так просто это не работает. Прикрепляю картинку с реализацией и ошибкой. 

![photo_2022-08-11_01-37-23](https://user-images.githubusercontent.com/54573430/183990954-3f401cba-18e5-460f-a1a0-675cf7382e68.jpg)

Попробовал гуглить, там предлагается подобное решение ниже. Это актуальная практика?

```
  periodToNoun (period: string): string {
    return this.availablePeriod[period as keyof typeof this.availablePeriod];
  }
```